receiver/m365receiver/factory.go

```diff
@@ -0,0 +1,97 @@
+// Copyright observIQ, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package m365receiver // import "github.com/observiq/observiq-otel-collector/receiver/m365receiver"
+
+import (
+	"context"
+	"time"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/confighttp"
+	"go.opentelemetry.io/collector/consumer"
+	"go.opentelemetry.io/collector/receiver"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/observiq/observiq-otel-collector/receiver/m365receiver/internal/metadata"
+)
+
+const (
+	typeStr = "m365"
+)
+
+// NewFactory creates a factory for Microsoft Office 365 receiver.
+func NewFactory() receiver.Factory {
+	return receiver.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		receiver.WithMetrics(createMetricsReceiver, metadata.Stability),
+		receiver.WithLogs(createLogsReceiver, metadata.Stability),
+	)
+}
+
+func createDefaultConfig() component.Config {
+	return &Config{
+		ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+			CollectionInterval: 5 * time.Second,
```

Comment:
Make sure this is set to a significantly longer collection interval before merging.

Reply:
the README should also talk about the frequency of metrics being available (one data point per day) as well.